spec/examples/by_set_pos_spec.rb

```diff
@@ -0,0 +1,42 @@
+require File.dirname(__FILE__) + '/../spec_helper'
+
+module IceCube
+
+  describe MonthlyRule, 'BYSETPOS' do
+    it 'should behave correctly' do
+      schedule = IceCube::Schedule.from_ical "RRULE:FREQ=MONTHLY;COUNT=4;BYDAY=WE;BYSETPOS=4"
+      schedule.start_time = Time.new(2015, 5, 28, 12, 0, 0)
+      expect(schedule.occurrences_between(Time.new(2015, 01, 01), Time.new(2017, 01, 01))).to eq([
+                                                                                                     Time.new(2015,6,24,12,0,0),
+                                                                                                     Time.new(2015,7,22,12,0,0),
+                                                                                                     Time.new(2015,8,26,12,0,0),
+                                                                                                     Time.new(2015,9,23,12,0,0)
+                                                                                                 ])
+    end
+
+    context 'when billing occurs at the end of the month' do
+      it 'should return the correct end of the months dates' do
+        schedule = IceCube::Schedule.from_ical "RRULE:FREQ=MONTHLY;COUNT=4;BYMONTHDAY=28,29,30,31;BYSETPOS=-1"
+        schedule.start_time = Time.new(2022, 11, 1, 12, 0, 0)
+        expect(schedule.occurrences_between(Time.new(2022, 10, 01), Time.new(2023, 10, 31))).to eq([
+                                                                                                     Time.new(2022,11,30,12,0,0),
+                                                                                                     Time.new(2022,12,31,12,0,0),
+                                                                                                     Time.new(2023,01,31,12,0,0),
+                                                                                                     Time.new(2023,02,28,12,0,0)
+                                                                                                   ])
+      end
+    end
+
+  end
```

Comment:
Suggestion to increase clarity, and thus ease it writing tests:
```suggestion
  describe MonthlyRule, 'BYSETPOS' do
    subject(:schedule) { IceCube::Schedule.from_ical(from_ical) }

    before(:each) do
      schedule.start_time = schedule_start
    end

    let(:occurrences) { schedule.occurrences_between(from_time, to_time) }

    context "when the rule is the first 4 Wednesdays ..." do
      let(:from_ical)       { "RRULE:FREQ=MONTHLY;COUNT=4;BYDAY=WE;BYSETPOS=4" }
      let(:schedule_start)  { Time.new(2015, 5, 28, 12, 0, 0) }
      let(:from_time)       { Time.new(2015, 1, 1) }
      let(:to_time)         { Time.new(2017, 1, 1) }

      it "returns the first 4 Wednesdays ..." do
        expect(occurrences).to eq(
          [
            Time.new(2015, 6, 24, 12, 0, 0),
            Time.new(2015, 7, 22, 12, 0, 0),
            Time.new(2015, 8, 26, 12, 0, 0),
            Time.new(2015, 9, 23, 12, 0, 0)
          ]
        )
      end
    end

    context "when the rule is the first 4 last days of the month by set pos" do
      let(:from_ical) { "RRULE:FREQ=MONTHLY;COUNT=4;BYMONTHDAY=28,29,30,31;BYSETPOS=-1" }
      let(:schedule_start)  { Time.new(2022, 11, 1, 12, 0, 0) }
      let(:from_time)       { Time.new(2022, 10, 1) }
      let(:to_time)         { Time.new(2023, 10, 31) }

      it "returns the first 4 last days of the month by set pos" do
        expect(occurrences).to eq(
          [
            Time.new(2022, 11, 30, 12, 0, 0),
            Time.new(2022, 12, 31, 12, 0, 0),
            Time.new(2023, 1,  31, 12, 0, 0),
            Time.new(2023, 2,  28, 12, 0, 0)
          ]
        )
      end
    end
  end
```

Reply:
I'd like to see some more tests that show that the `BYSETPOS` parameter is really working rather than just being lucky(?) for the selected cases.  Something like:

* `"RRULE:FREQ=MONTHLY;COUNT=4;BYMONTHDAY=28,29,30,31;BYSETPOS=-2"`
* `"RRULE:FREQ=MONTHLY;COUNT=4;BYMONTHDAY=28,29,30,31;BYSETPOS=4"`
* `"RRULE:FREQ=MONTHLY;COUNT=4;BYDAY=WE;BYSETPOS=-1"`